### ISSUE TITLE
When a source fails, choco exits instead of moving to next source

### ISSUE THREAD

**Issue by maxgaulhofer** [chocolatey/choco#612]
The Chocolatey Client does dont fall back to another source when multiple sources are configured and the source with the lowest priority is down.

Snip Chocolatey.config:

```
  <sources>  
    <source id="chocolatey Apeldoorn" value="https://chocolatey02/NuGet/chocolatey" disabled="false" priority="1" />
    <source id="chocolatey Amsterdam" value="https://chocolatey01/NuGet/chocolatey" disabled="false" priority="2" />
  </sources>
```

snip Chocolatey.log:

```
2016-02-04 20:44:40,891 [DEBUG] - _ Chocolatey:ChocolateyListCommand - Normal Run Mode _
2016-02-04 20:44:40,891 [DEBUG] - Searching for package information
2016-02-04 20:44:40,907 [DEBUG] - Running list with the following filter = ''
2016-02-04 20:44:40,907 [DEBUG] - --- Start of List ---
2016-02-04 20:45:02,034 [ERROR] - Unable to connect to the remote server
2016-02-04 20:45:02,034 [DEBUG] - Exiting with 1
``
```

**Comment by ferventcoder** [comment]
This may already be logged as an issue.

**Comment by ferventcoder** [comment]
Perhaps not. I will update the title.

**Comment by ferventcoder** [comment]
Choco should report a warning, then move on to the next source. Or ignore failing sources.

**Comment by maxgaulhofer** [comment]
Any idea when this is fixxed? My boss is asking me for a status update? :-) Are we talking about days, weeks or months?

**Comment by ferventcoder** [comment]
The answer is that it depends. :)

We triage and prioritize things accordingly. Then we determine which version the fix should go in. I had this listed at 0.9.10.1, which would have been probably a month or so. However this could affect others as well, so I adjusted it to 0.9.10, which is the next version going out. So you are talking a matter of days or weeks now.

It sounds like you may require a SLA and higher prioritization of features/fixes, so you should consider a business license when those become available in Q2. For more information on what you might expect, see https://www.kickstarter.com/projects/ferventcoder/chocolatey-the-alternative-windows-store-like-yum/description and then be sure to sign up for the newsletter https://chocolatey.us8.list-manage.com/subscribe?u=86a6d80146a0da7f2223712e4&id=73b018498d where we will announce what is coming.

Another way to move things up in priority is to submit the fixes for this issue as a PR. That guarantees a quicker evaluation.

**Comment by dragon788** [comment]
I've been seeing a similar issue related to search. If I have multiple sources enabled (internal and community feed), if I search a package not present internally it immediately fails rather than checking the community feed. I also noticed that search doesn't appear to care about the priority, setting the internal to a high number (99) and removing and readding Chocolatey at 1 (to avoid the 0 logic) and it still reports a failure. Only by disabling the internal source am I able to search and install from the community feed again.

**Comment by ferventcoder** [comment]
any failing source will cause issues.

**Comment by ferventcoder** [comment]
This seems to be working now, but it needs verified.

**Comment by chriszuercher** [comment]
I encountered this issue in v0.10.11. So in my opinion it is not fixed and should consideredas a High_Priority bug. 
I tried to debug it, but could not go to the source of the issues. It looks like something in the nuget-chocolatey project, but I'm not able to build this project (and therefore not to debug it).

**Comment by chriszuercher** [comment]
Is this issue probably related to https://github.com/chocolatey/nuget-chocolatey/issues/14 ?

**Comment by ferventcoder** [comment]
Thank goodness, this is fixed! Will be released with Chocolatey v0.10.12.
